### PR TITLE
Decision controller

### PR DESCRIPTION
**Pull request by mick-warehime**

### CODE REVIEW

File: src/decision_controller.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        self.prompt = prompt
+        self.choice = -1
+        self.options_keys = [pg.K_0, pg.K_1, pg.K_2,
+                             pg.K_3, pg.K_4, pg.K_5,
+                             pg.K_6, pg.K_7, pg.K_8,
+                             pg.K_9]
+
+    def set_option(self, idx: int, option: str) -> None:
+
+        assert idx < 10, 'decision must have less than 10 options'
```
Comment (dvirk) [inline src/decision_controller.py]: as a general rule, for checking arguments in public API it's prefered to use if statements followed by raising exceptions. [Link](https://mail.python.org/pipermail/python-list/2013-November/660568.html)
So here I would write
```
if not (0<= idx < 10):
     raise ValueError('idx must be non-negative and less than 10. Got %d' %(idx,) )
```

Comment (dvirk) [inline src/decision_controller.py]: as a general rule, for checking arguments in public API it's prefered to use if statements followed by raising exceptions. [Link](https://mail.python.org/pipermail/python-list/2013-November/660568.html)
So here I would write
```
if not (0<= idx < 10):
     raise ValueError('idx must be non-negative and less than 10. Got %d' %(idx,) )
```

File: src/decision_controller.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        assert idx >= 0, 'decision must be a >= 0'
+
+        key = self.options_keys[idx]
+        self.bind(key, self.get_choice_function(idx))
+        self.options[idx] = option
+
+    def get_choice_function(self, key_idx: int) -> Callable[..., None]:
+        def choice_func() -> None:
+            self.choice = key_idx
```
Comment (dvirk) [inline src/decision_controller.py]: why not use set_choice(key_idx) here.

File: src/decision_controller.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import controller
+from typing import Dict, Callable
+import pygame as pg
+
+
+class DecisionController(controller.Controller):
```
Comment (dvirk) [inline src/decision_controller.py]: Could you add a one line docstring explaining what this class is supposed to do?

File: src/test/pygame_mock.py [hunk 4]
```diff
@@ -11,5 +11,5 @@ def get_pressed(self) -> List[int]:
 
 class Pygame(object):
     def __init__(self) -> None:
-        self.key = Key(n_keys=10)
-        self.mouse = Key(n_keys=10)
+        self.key = Key(n_keys=500)
```
Comment (dvirk) [inline src/test/pygame_mock.py]: 500 seems pretty big. Why not 50?

File: src/test/test_controller.py [hunk 5]
```diff
@@ -173,3 +174,50 @@ def test_only_handle(self) -> None:
...
+        dc.set_option(2, 'two')
+
+        key1 = controller.pg.K_1
+        pg.key.pressed[key1] = 1
+
+        dc.update()
+
+        self.assertEqual(dc.choice, 1)
+
+    def test_set_option_2(self) -> None:
```
Comment (dvirk) [inline src/test/test_controller.py]: these three tests can be combined into one in a for loop, no?
In general each test should be looking at a distinct behavior of the code.

File: src/decision_controller.py (hunk 1 above)
Comment (mick-warehime) [reply src/decision_controller.py]: resolved

File: src/decision_controller.py (hunk 2 above)
Comment (mick-warehime) [reply src/decision_controller.py]: resolved

File: src/decision_controller.py (hunk 3 above)
Comment (mick-warehime) [reply src/decision_controller.py]: resolved

File: src/test/pygame_mock.py (hunk 4 above)
Comment (mick-warehime) [reply src/test/pygame_mock.py]: the actual length of this thing is something like 300 - 500 is fine

File: src/test/test_controller.py (hunk 5 above)
Comment (mick-warehime) [reply src/test/test_controller.py]: resolved

**Pull request merged**